### PR TITLE
Unmute testIndexingWithPrimaryOnBwcNodes

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description
https://github.com/opensearch-project/OpenSearch/pull/9686 temporarily muted `testIndexingWithPrimaryOnBwcNodes` due to addition of new field (StoreFileMetadata map) in ReplicationCheckpoint class. As the change is backported to 2.x, we can unmute testIndexingWithPrimaryOnBwcNodes test.

### Related Issues
Resolves #None
Relates: #9685 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
@Rishikesh1159 @mch2 @reta @kotwanikunal : Request for review

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@dreamer-89 you may need to rebase sadly

**Review by kkmr — APPROVED** [review]

**Comment by dreamer-89** [comment]
> @dreamer-89 you may need to rebase sadly

Thanks @reta. I rebased my branch with main. Waiting for green gradle check before merge.

**Pull request merged**
